### ISSUE TITLE
For.a MFR-8000

### ISSUE THREAD

**Issue by sstubblefield** [bitfocus/companion-module-requests#323]
I'd love to be able to either control Input/Output routing or select Salvo's (presets) in the for.a router.

**Comment by josephdadams** [comment]
Any protocol or documentation information you can share on this FOR-A router you want to control?

**Comment by sstubblefield** [comment]
@josephdadams I've reached out to For.A to see if there's any information on it. I'll update when I know more.

**Comment by sstubblefield** [comment]
From For.A:

"MFR-8000 supports to control via Serial / LAN command.
Communication protocol is TCP/IP, which is based on HARRIS.
For detail, please refer to the "6. Serial / LAN Command Control" on the manual. You can find the commands.
If you can find the protocol for stream deck, you may be able to send a command from it."

@josephdadams I've attached the manual here. Let me know if this is something the Stream Deck can do.

[MFR8000-E-E8R1_SCS.pdf](https://github.com/bitfocus/companion-module-requests/files/5631775/MFR8000-E-E8R1_SCS.pdf)

**Comment by josephdadams** [comment]
Yes, this can be used to create a module. So now you have a few options:

- Wait and see if anyone wants to create this for you and/or themselves
- Learn some code and make the module (I can point you in the direction of some modules to consider as boilerplates/templates for this)
- Offer a financial incentive to have someone create this for you

**Comment by sstubblefield** [comment]
@josephdadams Okay cool. I might take a look at it myself. I'm a former JS dev, so I'd be interested in checking out some boilerplates/templates. Thanks for your help.

**Comment by sstubblefield** [comment]
@josephdadams where could I find the boilerplates/templates? I dug around a little bit, but wasn't sure if they were public or not.

**Comment by josephdadams** [comment]
It looks to be a simple telnet ascii protocol, so take a look at any of the Roland modules.

**Comment by josephdadams** [comment]
#354 related
